packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -387,18 +452,20 @@ export class QueuedRequestController extends BaseController<
     }
 
     try {
+      let endRequest: (() => void) | undefined;
       // Queue request for later processing
       // Network switch is handled when this batch is processed
       if (
         this.state.queuedRequestCount > 0 ||
         this.#originOfCurrentBatch !== request.origin ||
-        this.#networkClientIdOfCurrentBatch !== request.networkClientId
+        this.#networkClientIdOfCurrentBatch !== request.networkClientId ||
+        (this.#processingRequestCount > 0 &&
```

Comment:
Nit: Maybe worth an additional comment here to explain why these are a special case

Reply:
https://github.com/MetaMask/core/pull/4846/commits/65e0cba9a49c4c7f52775bfcfb3a0ce29551028a